Source/VMware.PSDesiredStateConfiguration/Tests/Integration/Invoke-VmwDscConfiguration.Integration.Tests.ps1

```diff
@@ -0,0 +1,160 @@
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+$root = Split-Path (Split-Path $PSScriptRoot)
+
+$module = Join-Path $root 'VMware.PSDesiredStateConfiguration.psd1'
+
+Import-Module $module -Force
+
+InModuleScope -ModuleName 'VMware.PSDesiredStateConfiguration' {
+    Describe 'Invoke-VmwDscConfiguration integration tests' {
+        try {
+            $root = $PSScriptRoot
+            $serverConfigPath = Join-Path $root 'ServerConfig.ps1'
+    
+            # server data is retrieved from a ServerConfig.ps1 file
+            $serverConfigs = . $serverConfigPath
+    
+            foreach ($serverConfig in $serverConfigs) {
+                # establish connection to the vCenter
+                $Script:Connection = Connect-ViServer -Server $serverConfig.Server -User $serverConfig.User -Password $serverConfig.Password
+            }
+    
+            $Script:ConfigData = @{
+                AllNodes = @(
+                    foreach ($serverConfig in $serverConfigs) {
+                        @{
+                            NodeName = $serverConfig.Server
+                        }
+                    }
+                )
+                DatacenterFolder = @{
+                    Name = 'MyDatacentersFolder'
+                    Location = ''
+                }
+                Datacenter = @{
+                    Name = 'MyDatacenter'
+                    Location = 'MyDatacentersFolder'
+                }
+            }
+    
+            $root = $PSScriptRoot
+    
+            $configFolder = Join-Path $root 'Configurations'
+    
+            $Script:ConfigPath = Join-Path $configFolder 'vSphereNodeConfiguration.ps1'
+    
+            $Script:DscConfigurationObj = $null
+    
+            It 'Should compile correctly' {
+                # Arrange
+                $splatParams = @{
+                    ConfigName = 'Config_Add'
+                    ConfigurationData = $Script:ConfigData
+                }
+    
+                . $Script:ConfigPath
+    
+                # Act
+                $Script:DscConfigurationObj = New-VmwDscConfiguration @splatParams
+    
+                # Assert
+                $Script:DscConfigurationObj | Should -Not -Be $null
```

Comment:
Additional asserts could be added here to check that the returned object is correctly populated with the **Node** and **DSC Resources** data.